gcloud-java-vision/src/main/java/com/google/cloud/vision/spi/v1/ImageAnnotatorApi.java

```diff
@@ -0,0 +1,222 @@
+    return new ImageAnnotatorApi(settings);
+  }
+
+  /**
+   * Constructs an instance of ImageAnnotatorApi, using the given settings.
+   * This is protected so that it easy to make a subclass, but otherwise, the static
+   * factory methods should be preferred.
+   */
+  protected ImageAnnotatorApi(ImageAnnotatorSettings settings) throws IOException {
+    this.settings = settings;
+    this.executor = settings.getExecutorProvider().getOrBuildExecutor();
+    this.channel = settings.getChannelProvider().getOrBuildChannel(this.executor);
+
+    this.batchAnnotateImagesCallable =
+        ApiCallable.create(settings.batchAnnotateImagesSettings(), this.channel, this.executor);
+
+    if (settings.getChannelProvider().shouldAutoClose()) {
+      closeables.add(
+          new Closeable() {
+            @Override
+            public void close() throws IOException {
+              channel.shutdown();
+            }
+          });
+    }
+    if (settings.getExecutorProvider().shouldAutoClose()) {
+      closeables.add(
+          new Closeable() {
+            @Override
+            public void close() throws IOException {
+              executor.shutdown();
+            }
+          });
+    }
+  }
+
+  // AUTO-GENERATED DOCUMENTATION AND METHOD
+  /**
+   * Run image detection and annotation for a batch of images.
+   *
+   * Sample code:
+   * <pre><code>
+   * try (ImageAnnotatorApi imageAnnotatorApi = ImageAnnotatorApi.create()) {
+   *   List&lt;AnnotateImageRequest&gt; requests = new ArrayList&lt;&gt;();
+   *   BatchAnnotateImagesResponse response = imageAnnotatorApi.batchAnnotateImages(requests);
+   * }
+   * </code></pre>
+   *
+   * @param requests Individual image annotation requests for this batch.
+   * @throws com.google.api.gax.grpc.ApiException if the remote call fails
+   */
+  public final BatchAnnotateImagesResponse batchAnnotateImages(
+      List<AnnotateImageRequest> requests) {
+    BatchAnnotateImagesRequest request =
+        BatchAnnotateImagesRequest.newBuilder().addAllRequests(requests).build();
+    return batchAnnotateImages(request);
+  }
+
+  // AUTO-GENERATED DOCUMENTATION AND METHOD
+  /**
+   * Run image detection and annotation for a batch of images.
+   *
+   * Sample code:
+   * <pre><code>
+   * try (ImageAnnotatorApi imageAnnotatorApi = ImageAnnotatorApi.create()) {
+   *   List&lt;AnnotateImageRequest&gt; requests = new ArrayList&lt;&gt;();
+   *   BatchAnnotateImagesRequest request = BatchAnnotateImagesRequest.newBuilder()
+   *     .addAllRequests(requests)
+   *     .build();
+   *   BatchAnnotateImagesResponse response = imageAnnotatorApi.batchAnnotateImages(request);
+   * }
+   * </code></pre>
+   *
+   * @param request The request object containing all of the parameters for the API call.
+   * @throws com.google.api.gax.grpc.ApiException if the remote call fails
+   */
+  private final BatchAnnotateImagesResponse batchAnnotateImages(
```

Comment:
Yes, our code generator automatically hides the request-based methods when the request has only one field, so the flattened version is more direct. Will this cause a problem? We can re-evaluate.